### PR TITLE
Appropriately map between buffer and view spans in our completion controller

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
test vsi please

**Comment by CyrusNajmabadi** [comment]
retest vsi please

**Comment by CyrusNajmabadi** [comment]
retest windows_eta_open_prtest  please

**Comment by CyrusNajmabadi** [comment]
tagging @Pilchie @rchande @dotnet/roslyn-ide 

Please review.  This is the change to fix the bad span mapping affecting Razor completion

**Comment by CyrusNajmabadi** [comment]
retest windows_eta_open_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p1_open_prtest  please

**Comment by CyrusNajmabadi** [comment]
retest windows_vsi_p2_open_prtest  please

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs [hunk 1]
```diff
@@ -95,23 +100,38 @@ private void Commit(PresentationItem item, Model model, char? commitChar, Cancel
...
+                            textChangeInBuffer.Span.ToSpan());
+
+                        var viewSpan = bufferGraph.GetSubjectBufferTextSpanInViewBuffer(
+                            currentSpanInBuffer.Span.ToTextSpan());
+
+                        textChangesInView.Add(new TextChange(viewSpan.TextSpan, textChangeInBuffer.NewText));
+                    }
+
+                    // Note: we currently create the edit on the textview's buffer.  This is 
+                    // necessary as our own
```
Comment (rchande) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs]: our own what?

**Comment by rchande** [comment]
:+1:

**Comment by Pilchie** [comment]
retest windows_eta_open_prtest please
retest windows_vsi_p2_open_prtest please

The failure in vsi_p1 seems quite plausibly related to me.

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs (hunk 1 above)
Comment (mattwar) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_Commit.cs]: The handwriting just trails off at that point.  Do you think cyrus is in trouble?

**Comment by CyrusNajmabadi** [comment]
@dotnet-bot retest windows_eta_open_prtest please
// Previous failure: http://dotnet-ci.cloudapp.net/job/Private/job/dotnet_roslyn-internal/job/microupdate/job/windows_eta_open_prtest/43/
// Retest reason:

**Comment by CyrusNajmabadi** [comment]
> The failure in vsi_p1 seems quite plausibly related to me.
Turns out is a U3 regression in the editor snippet picker.

**Comment by CyrusNajmabadi** [comment]
@rchande @Pilchie do you think you'll be able to smoke test this out?

**Comment by rchande** [comment]
@CyrusNajmabadi sure. I'll focus specifically on debugger intellisense scenarios.

**Comment by Pilchie** [comment]
@rchande We should look at all projection related scenarios. At least:

1. Razor (cshtml/vbhtml)
2. Webforms (aspx)
3. Expression Editor
4. Diff View
5 Debugger IntelliSense

Additionally, we should try to look at things where brace completion, formatting and completion come together and intersect.

**Comment by Pilchie** [comment]
Once @rchande has played with it a bit, lets build vsix to give to the ASP.NET team for some buddy testing too.

**Comment by Pilchie** [comment]
:+1:

**Comment by rchande** [comment]
@Pilchie @CyrusNajmabadi What is supposed to happen with ABC if you type
`string.Join<`, committing `Join` with `<`? We don't start a brace completion session for the `<` when it commits completion. I see this with Cyrus' changes, but also with my current dogfooding bits (indicating that Cyrus didn't regress it, if it is indeed a regression).

**Comment by rchande** [comment]
Apart from the above question, I didn't find any issues testing 3,4,5. I'll let the web team provide their expertise for web scenarios.

**Comment by Pilchie** [comment]
@rchande That looks like https://github.com/dotnet/roslyn/issues/2173.

**Comment by Pilchie** [comment]
Either @rchande or @CyrusNajmabadi  can you build a RoslynDeployment.vsix with these changes to give the web team?

**Comment by tannergooding** [comment]
test windows_eta_open_prtest please
test windows_vsi_p1_open_prtest please
test windows_vsi_p2_open_prtest please

**Pull request merged**